From dad60d44960c412738b2cfd30f96e4b1d9767362 Mon Sep 17 00:00:00 2001
From: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Date: Fri, 3 Jan 2020 05:26:30 +0530
Subject: [PATCH] FROMGIT: UBUNTU: SAUCE: ASoC: SOF: Intel: add Build support for
 Skylake and Kabylake

BugLink: https://bugs.launchpad.net/bugs/1826181

Add Skylake and Kabylake to Kconfig and Makefile

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
(cherry picked from commit ae49238438893d9fdc8b439da68c5f7d23ffb4a3
git://github.com/thesofproject/linux.git)
Signed-off-by: Hui Wang <hui.wang@canonical.com>
Signed-off-by: Timo Aaltonen <timo.aaltonen@canonical.com>
---
 sound/soc/sof/intel/Kconfig  | 32 ++++++++++++++++++++++++++++++++
 sound/soc/sof/intel/Makefile |  4 ++--
 2 files changed, 34 insertions(+), 2 deletions(-)

diff --git a/sound/soc/sof/intel/Kconfig b/sound/soc/sof/intel/Kconfig
index 6236ee524251..dd9ee6b447f9 100644
--- a/sound/soc/sof/intel/Kconfig
+++ b/sound/soc/sof/intel/Kconfig
@@ -24,6 +24,8 @@ config SND_SOC_SOF_INTEL_PCI
 	select SND_SOC_SOF_GEMINILAKE  if SND_SOC_SOF_GEMINILAKE_SUPPORT
 	select SND_SOC_SOF_CANNONLAKE  if SND_SOC_SOF_CANNONLAKE_SUPPORT
 	select SND_SOC_SOF_COFFEELAKE  if SND_SOC_SOF_COFFEELAKE_SUPPORT
+	select SND_SOC_SOF_KABYLAKE    if SND_SOC_SOF_KABYLAKE_SUPPORT
+	select SND_SOC_SOF_SKYLAKE     if SND_SOC_SOF_SKYLAKE_SUPPORT
 	select SND_SOC_SOF_ICELAKE     if SND_SOC_SOF_ICELAKE_SUPPORT
 	select SND_SOC_SOF_COMETLAKE_LP if SND_SOC_SOF_COMETLAKE_LP_SUPPORT
 	select SND_SOC_SOF_COMETLAKE_H if SND_SOC_SOF_COMETLAKE_H_SUPPORT
@@ -183,6 +185,36 @@ config SND_SOC_SOF_COFFEELAKE
 	  This option is not user-selectable but automagically handled by
 	  'select' statements at a higher level
 
+config SND_SOC_SOF_KABYLAKE_SUPPORT
+	bool "SOF support for Kabylake"
+	help
+	  This adds support for Sound Open Firmware for Intel(R) platforms
+	  using the Kabylake processors.
+	  Say Y if you have such a device.
+	  If unsure select "N".
+
+config SND_SOC_SOF_KABYLAKE
+	tristate
+	select SND_SOC_SOF_HDA_COMMON
+	help
+	  This option is not user-selectable but automagically handled by
+	  'select' statements at a higher level
+
+config SND_SOC_SOF_SKYLAKE_SUPPORT
+	bool "SOF support for Skylake"
+	help
+	  This adds support for Sound Open Firmware for Intel(R) platforms
+	  using the Skylake processors.
+	  Say Y if you have such a device.
+	  If unsure select "N".
+
+config SND_SOC_SOF_SKYLAKE
+	tristate
+	select SND_SOC_SOF_HDA_COMMON
+	help
+	  This option is not user-selectable but automagically handled by
+	  'select' statements at a higher level
+
 config SND_SOC_SOF_ICELAKE_SUPPORT
 	bool "SOF support for Icelake"
 	help
diff --git a/sound/soc/sof/intel/Makefile b/sound/soc/sof/intel/Makefile
index 2caf14d32f78..84a65c80f62a 100644
--- a/sound/soc/sof/intel/Makefile
+++ b/sound/soc/sof/intel/Makefile
@@ -8,8 +8,8 @@ snd-sof-intel-ipc-objs := intel-ipc.o
 
 snd-sof-intel-hda-common-objs := hda.o hda-loader.o hda-stream.o hda-trace.o \
 				 hda-dsp.o hda-ipc.o hda-ctrl.o hda-pcm.o \
-				 hda-dai.o hda-bus.o \
-				 apl.o cnl.o
+				 hda-dai.o hda-bus.o hda-loader-skl.o \
+				 skl.o apl.o cnl.o
 
 snd-sof-intel-hda-objs := hda-codec.o
 
-- 
2.17.1

